src/services/auth/guestLogin.tsx

```diff
@@ -0,0 +1,25 @@
+import axios from "axios";
+
+const handleGuestLogin = async (
+    setIsUserLoggedIn: React.Dispatch<React.SetStateAction<boolean>>,
```

Comment:
didn't understand this, can  you please explain?

Reply:
it was a setter function that was being passed as an argument to the guest login handler and therefore provided its type. And have provided the setter function type as a boolean value. And hence its type is restricted to only boolean values this way, if I would by mistake assign it a different type then it would throw me an error.

Hope this answers what you asked!!